aeneas/src/ir/VariantSolver.v3

```diff
@@ -83,6 +101,197 @@ def ON_STACK = -1;
 def EMPTY_FIELD = VariantField.new(null, []);
 def NO_FIELDS = Array<VariantField>.new(0);
 
+class VariantSolver(vnorm: VariantNormalizer, usePacking: bool, useTagScalar: bool) {
+	var normFields: Array<Array<Type>>;
+
+	var fields: Vector<CaseField>; // an ordering of fields to solve for during packing
+
+	// final solutions
+	var scalarAsns: HashMap<CaseField, int>; // field location -> scalar idx
```

Comment:
Any reason not to combine the scalar assignment and interval assignment into one hashmap?

Reply:
No reason, I'll combine them.